processors/communicationdetails-deriver/src/test/java/org/hawkular/apm/processor/communicationdetails/CommunicationDetailsDeriverTest.java

```diff
@@ -473,11 +473,11 @@ public void testProcessSingleWithClient() {
         trace2.setHostName("host2");
         trace2.setHostAddress("addr2");
         trace2.setPrincipal("p1");
-        trace2.getProperties().add(new Property("prop1", "value1"));
 
         Consumer c2 = new Consumer();
         c2.setUri("TheURI");
         c2.setDuration(1200000000);
+        c2.getProperties().add(new Property("prop1", "value1"));
```

Comment:
Can you also add a test with a Property with a null value, to test the condition mentioned on a previous comment?